internal/pkg/template/template_test.go

```diff
@@ -182,3 +183,116 @@ func TestTemplate_Parse(t *testing.T) {
 		})
 	}
 }
+
+func TestTemplate_ParseNetwork(t *testing.T) {
+	type cfn struct {
+		Resources struct {
+			Service struct {
+				Properties struct {
+					NetworkConfiguration map[interface{}]interface{} `yaml:"NetworkConfiguration"`
+				} `yaml:"Properties"`
+			} `yaml:"Service"`
+		} `yaml:"Resources"`
+	}
+
+	testCases := map[string]struct {
+		input *NetworkOpts
+
+		wantedNetworkConfig string
+	}{
+		"should render AWS VPC configuration for public subnets by default": {
+			input: nil,
+			wantedNetworkConfig: `
+  AwsvpcConfiguration:
+    AssignPublicIp: ENABLED
+    Subnets:
+    - Fn::Select:
+      - 0
+      - Fn::Split:
+        - ','
+        - Fn::ImportValue: !Sub '${AppName}-${EnvName}-PublicSubnets'
+    - Fn::Select:
+      - 1
+      - Fn::Split:
+        - ','
+        - Fn::ImportValue: !Sub '${AppName}-${EnvName}-PublicSubnets'
+    SecurityGroups:
+      - Fn::ImportValue: !Sub '${AppName}-${EnvName}-EnvironmentSecurityGroup'
+`,
+		},
+		"should render AWS VPC configuration for private subnets": {
+			input: &NetworkOpts{
+				AssignPublicIP: "DISABLED",
+				SubnetsType:    "PrivateSubnets",
+			},
+			wantedNetworkConfig: `
+  AwsvpcConfiguration:
+    AssignPublicIp: DISABLED
+    Subnets:
+    - Fn::Select:
+      - 0
+      - Fn::Split:
+        - ','
+        - Fn::ImportValue: !Sub '${AppName}-${EnvName}-PrivateSubnets'
+    - Fn::Select:
+      - 1
+      - Fn::Split:
+        - ','
+        - Fn::ImportValue: !Sub '${AppName}-${EnvName}-PrivateSubnets'
+    SecurityGroups:
+      - Fn::ImportValue: !Sub '${AppName}-${EnvName}-EnvironmentSecurityGroup'
+`,
+		},
+		"should render AWS VPC configuration for private subnets with security groups": {
+			input: &NetworkOpts{
+				AssignPublicIP: "DISABLED",
+				SubnetsType:    "PrivateSubnets",
+				SecurityGroups: []string{
+					"sg-1bcf1d5b",
+					"sg-asdasdas",
+				},
+			},
+			wantedNetworkConfig: `
+  AwsvpcConfiguration:
+    AssignPublicIp: DISABLED
+    Subnets:
+    - Fn::Select:
+      - 0
+      - Fn::Split:
+        - ','
+        - Fn::ImportValue: !Sub '${AppName}-${EnvName}-PrivateSubnets'
+    - Fn::Select:
+      - 1
+      - Fn::Split:
+        - ','
+        - Fn::ImportValue: !Sub '${AppName}-${EnvName}-PrivateSubnets'
+    SecurityGroups:
+      - Fn::ImportValue: !Sub '${AppName}-${EnvName}-EnvironmentSecurityGroup'
+      - "sg-1bcf1d5b"
+      - "sg-asdasdas"
+`,
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			// GIVEN
+			tpl := New()
+			wanted := make(map[interface{}]interface{})
+			err := yaml.Unmarshal([]byte(tc.wantedNetworkConfig), &wanted)
+			require.NoError(t, err, "unmarshal wanted config")
+
+			// WHEN
+			content, err := tpl.ParseLoadBalancedWebService(WorkloadOpts{
```

Comment:
Do you think we should put this test to `workload_test.go` instead?

Reply:
It might not be possible because `ParseLoadBalancedWebService` actually uses the Go template under `templates/workloads/services/lb-web/cf.yml`

Reply:
Done in the latest revision.